pkg/cmd/codespace/create.go

```diff
@@ -113,12 +114,17 @@ func (a *App) Create(ctx context.Context, opts createOptions) error {
 
 	promptForRepoAndBranch := userInputs.Repository == ""
 	if promptForRepoAndBranch {
+		var defaultRepo string
+		if currentRepo, err := a.baseRepo(); err == nil {
+			defaultRepo = ghrepo.FullName(currentRepo)
+		}
```

Comment:
I see two UX issues here both stemming from using `SmartBaseRepoFunc` to determine the base repo:

1. `SmartBaseRepoFunc` outputs error messages to stderr which seems like something we would want to silence in this case since we are only using the resulting value if there is no error. I would have to double check but I believe all other use cases of `SmartBaseRepoFunc` we exit on error so that the user can act on the error message from that function.
2. The `SmartBaseRepoFunc` has cases where it will return a repo that has not actually been created on GitHub. Having that set as the default and being selected would likely cause an error, right? Having the default value cause an error seems like something we would want to avoid.

Having said that, I think number 1 is a much more glaring issue that will cause confusion. 

A different approach that we can take is to use the `Remotes()` factory method which will return a list of `git` remotes, and from there use the `ResolvedRemote()` method and use the as the default value. The `ResolvedRemote()` method will return the remote that was selected with the `repo set-default` function. This approach would solve both issues as there is no output from either of those methods and by using the remote that the user has already selected as the default we know that it exists on GitHub. This is the approach we use in `repo set-default` command.

Reply:
Thanks for the review, great point. I've changed the implementation as you suggest.